src/main/scala/cromwell/engine/CallExecutionActor.scala

```diff
@@ -55,19 +54,12 @@ class CallExecutionActor(backendCall: BackendCall) extends Actor with CromwellAc
    * Schedule work according to the schedule of the `backoff`.
    */
   private def scheduleWork(work: => Unit): Unit = {
-    val interval = backoff.nextBackOffMillis().millis
-    context.system.scheduler.scheduleOnce(interval) {
+    val interval = backendCall.pollBackoff.next.backoffMillis
+    context.system.scheduler.scheduleOnce(interval.millis) {
       work
     }
```

Comment:
I think you're right -_- This stuff is inherently mutable I think though. We're going to end up with a `var` somewhere.
I still prefer this ADT implementation to the one before, we could just add a `var backoff` in the CallExecutionActor and do `backoff = backoff.next` after calling `backoff.backoffMillis`...
It works for recursive methods  though (like in `TryUtil.withRetry`) where we can pass `backoff.next` to the next loop.